consensus/signing/signing.go

```diff
@@ -43,40 +39,31 @@ func NewSignatureHolder(args ArgsSignatureHolder) (*signatureHolder, error) {
 
 	sigSharesSize := uint16(len(args.PubKeys))
 	sigShares := make([][]byte, sigSharesSize)
-	pk, err := convertStringsToPubKeysBytes(args.PubKeys)
+
+	pybKeysBytes, err := convertStringsToPubKeysBytes(args.PubKeys)
 	if err != nil {
 		return nil, err
 	}
 
 	data := &signatureHolderData{
-		pubKeys:   pk,
-		privKey:   args.PrivKey,
+		pubKeys:   pybKeysBytes,
+		privKey:   args.PrivKeyBytes,
 		sigShares: sigShares,
 	}
 
-	multiSigner, err := args.MultiSignerContainer.GetMultiSigner(0)
-	if err != nil {
-		return nil, err
-	}
-
 	return &signatureHolder{
 		data:                 data,
 		mutSigningData:       sync.RWMutex{},
-		singleSigner:         args.SingleSigner,
 		multiSignerContainer: args.MultiSignerContainer,
-		multiSigner:          multiSigner,
 		keyGen:               args.KeyGenerator,
 	}, nil
 }
 
 func checkArgs(args ArgsSignatureHolder) error {
-	if check.IfNil(args.SingleSigner) {
-		return ErrNilSingleSigner
-	}
 	if check.IfNil(args.MultiSignerContainer) {
 		return ErrNilMultiSignerContainer
 	}
-	if check.IfNil(args.PrivKey) {
+	if args.PrivKeyBytes == nil {
```

Comment:
can you check the len instead? 
it can be len 0 but not nil

Reply:
done